pkg/storage/utils/indexer/index/autoincrement.go

```diff
@@ -75,17 +75,53 @@ func (idx *Autoincrement) Init() error {
 
 // Lookup exact lookup by value.
 func (idx *Autoincrement) Lookup(v string) ([]string, error) {
-	searchPath := path.Join(idx.indexRootDir, v)
-	oldname, err := idx.storage.ResolveSymlink(context.Background(), searchPath)
-	if err != nil {
-		if os.IsNotExist(err) {
-			err = &idxerrs.NotFoundErr{TypeName: idx.typeName, IndexBy: idx.indexBy, Value: v}
+	return idx.LookupCtx(context.Background(), v)
+}
+
+// LookupCtx retieves multiple exact values and allows passing in a context
+func (idx *Autoincrement) LookupCtx(ctx context.Context, values ...string) ([]string, error) {
+	var allValues []string
+	var err error
+	if len(values) != 1 {
+		allValues, err = idx.storage.ReadDir(context.Background(), path.Join("/", idx.indexRootDir))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		allValues = values
+	}
+
+	var matches = []string{}
+	for _, av := range allValues {
```

Comment:
It could be more efficient to turn allValues into a `map[string]struct{}` to have faster lookups.